src/appleseed/renderer/kernel/lighting/directlightingintegrator.cpp

```diff
@@ -670,13 +676,32 @@ void DirectLightingIntegrator::add_non_physical_light_sample_contribution(
     // Evaluate the light.
     Vector3d emission_position, emission_direction;
     Spectrum light_value(Spectrum::Illuminance);
-    light->evaluate(
+
+    // Generate a uniform sample in [0,1).
+    sampling_context.split_in_place(2, 1);
+    const Vector2d s = sampling_context.next2<Vector2d>();
```

Comment:
That's correct, here is the details of how this works:

    sampling_context.split_in_place(2, 1);

This creates a new "trajectory" or "sampling sequence": the 2 means "2D samples" and the 1 means that we want a single sample in that sequence.

    sampling_context.next2<Vector2d>();

This returns the next (and in this case, the only) 2D sample (of type Vector2d, i.e. as two double-precision values). The 2 in `next2()` means that we want a semi-open intervals for samples: [0, 1) (closed on the left, open on the right, i.e. 0 is part of the interval but 1 isn't). That's the only variant available with sampling contexts since it's the only one that we need so far.

With pure random numbers, we have two additional variants: Functions suffixed with 1 such as `rand_double1()` return values in the interval [0, 1] (closed on both ends) and functions suffixed with 3 such as `rand_double3()` return values in the interval (0, 1) (open on both ends).